tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_withdrawals.py

```diff
@@ -94,14 +94,351 @@ def test_pending_withdrawals_one_skipped_one_effective(spec, state):
+    # so the validator is partially withdrawable after pending withdrawal is processed
+    state.balances[validator_index] = spec.MAX_EFFECTIVE_BALANCE_ELECTRA + spec.EFFECTIVE_BALANCE_INCREMENT * 3
+    assert spec.is_partially_withdrawable_validator(
+        state.validators[validator_index],
+        state.balances[validator_index] - pending_withdrawal_0.amount - pending_withdrawal_1.amount
+    )
+
+    next_slot(spec, state)
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(
+        spec, state,
+        execution_payload,
+        num_expected_withdrawals=3,
+        fully_withdrawable_indices=[],
+        partial_withdrawals_indices=[validator_index],
+        pending_withdrawal_requests=[pending_withdrawal_0, pending_withdrawal_1]
+    )
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_with_sweep_different_validator(spec, state):
+    # Ensure validator will be processed by the sweep
+    validator_index_0 = min(len(state.validators), spec.MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP) // 2 - 1
+    validator_index_1 = min(len(state.validators), spec.MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP) // 2
+
+    # Initiate pending withdrawal for the first validator
+    pending_withdrawal_0 = prepare_pending_withdrawal(
+        spec, state,
+        validator_index_0,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        amount=spec.EFFECTIVE_BALANCE_INCREMENT
+    )
+
+    # Make the second validator partially withdrawable by the sweep
+    set_compounding_withdrawal_credential_with_balance(
+        spec, state, validator_index_1,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+        balance=(spec.MAX_EFFECTIVE_BALANCE_ELECTRA + spec.EFFECTIVE_BALANCE_INCREMENT)
+    )
+
+    assert spec.is_partially_withdrawable_validator(
+        state.validators[validator_index_1],
+        state.balances[validator_index_1]
+    )
+
+    next_slot(spec, state)
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(
+        spec, state,
+        execution_payload,
+        num_expected_withdrawals=2,
+        fully_withdrawable_indices=[],
+        partial_withdrawals_indices=[validator_index_1],
+        pending_withdrawal_requests=[pending_withdrawal_0]
+    )
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_mixed_with_sweep_and_fully_withdrawable(spec, state):
+    num_full_withdrawals = spec.MAX_WITHDRAWALS_PER_PAYLOAD // 4
+    num_partial_withdrawals = spec.MAX_WITHDRAWALS_PER_PAYLOAD // 4
+    num_full_withdrawals_comp = spec.MAX_WITHDRAWALS_PER_PAYLOAD // 4
+    num_partial_withdrawals_comp = spec.MAX_WITHDRAWALS_PER_PAYLOAD // 4
+    num_pending_withdrawal_requests = spec.MAX_WITHDRAWALS_PER_PAYLOAD // 4
```

Comment:
I'm a little confused here. There are 5 counts, each is a 4th of the max. The first 4 make total sense; you're splitting the withdrawals up. I'm just confused by the last line. Does `num_pending_withdrawal_requests` need to be `// 4`? How is this influenced by the others? Could it be `// 2` instead?

Reply:
If it was `// 2` then it would equal `MAX_PENDING_PARTIALS_PER_WITHDRAWALS_SWEEP` which isn’t the desired setup for this test case. But it rather should be a fraction of `MAX_PENDING_PARTIALS_PER_WITHDRAWALS_SWEEP`